src/Microsoft.Health.Api/Features/Audit/AuditEventTypeMapping.cs

```diff
@@ -38,7 +38,7 @@ public class AuditEventTypeMapping : IAuditEventTypeMapping, IHostedService
 {
     private readonly IActionDescriptorCollectionProvider _actionDescriptorCollectionProvider;
 
-    private IReadOnlyDictionary<(string ControllerName, string ActionName), Attribute> _attributeDictionary;
```

Comment:
I see, in some place you use ImmutableDictionary compared to Dictionary. It is because that was a static property?

Reply:
It was because it is a `private` member and we don't necessarily care about its immutability as much, as only the instance has access. But there could be an argument for making it an `ImmutableDictionary<TKey, TValue>`